### DISCUSSION TITLE
Image without LibreOffice

### DISCUSSION THREAD

**Discussion by Tobias-08** [gotenberg/gotenberg#929] [Q&A] [answered]
Hi,

we are only using Gotenberg's Chromium-html-route to convert html. Now I tried to build a smaller Gotenberg-image without LibreOffice and the other unneeded pdf-tools. I hoped to disable the other features by:

```
--libreoffice-disable-routes=true --pdfengines-disable-routes=true --webhook-disable=true
```

But this does not seem to work as the modified image still checks the existence of these tools at startup and therefore fails:

```
Version: 8.8.1
-------------------------------------------------------
[SYSTEM] modules: api chromium exiftool libreoffice libreoffice-api libreoffice-pdfengine logging pdfcpu pdfengines pdftk prometheus qpdf webhook 
[FATAL] provision module api: get routers: provision module chromium: get PDF engine provider: get module: provision module pdfengines: get PDF engines: validate module exiftool: ExifTool binary path does not exist: stat /usr/bin/exiftool: no such file or directory
```

Is there a way to disable these checks?

Thanks!

**Comment by gulien** [comment, marked as answer]
Hello,

You must create your own `main` like this: https://github.com/gotenberg/gotenberg/blob/main/cmd/gotenberg/main.go.
And import the required modules - see the standard list https://github.com/gotenberg/gotenberg/blob/main/pkg/standard/imports.go.

**Reply by Tobias-08** [reply]
@gulien Thank you!
Gotenberg seems to need at least one pdf engine so I can't omit

```
_ "github.com/gotenberg/gotenberg/v8/pkg/modules/exiftool"
_ "github.com/gotenberg/gotenberg/v8/pkg/modules/pdfengines"
```

But leaving out libreoffice, pdftk, ... works!
